walletnode/services/artworkdownload/task.go

```diff
@@ -0,0 +1,152 @@
+package artworkdownload
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/common/service/task"
+	"github.com/pastelnetwork/gonode/common/service/task/state"
+	"github.com/pastelnetwork/gonode/walletnode/services/artworkdownload/node"
+)
+
+// Task is the task of downloading artwork.
+type Task struct {
+	task.Task
+	*Service
+
+	Ticket *Ticket
+	File   []byte
+	err    error
+}
+
+// Run starts the task
+func (task *Task) Run(ctx context.Context) error {
+	ctx = log.ContextWithPrefix(ctx, fmt.Sprintf("%s-%s", logPrefix, task.ID()))
+
+	task.SetStatusNotifyFunc(func(status *state.Status) {
+		log.WithContext(ctx).WithField("status", status).Debugf("States updated")
+	})
+
+	log.WithContext(ctx).Debugf("Start task")
+	defer log.WithContext(ctx).Debugf("End task")
+
+	if err := task.run(ctx); err != nil {
+		task.err = err
+		task.UpdateStatus(StatusTaskRejected)
+		log.WithContext(ctx).WithErrorStack(err).Warnf("Task is rejected")
+		return nil
+	}
+
+	task.UpdateStatus(StatusTaskCompleted)
+	return nil
+}
+
+func (task *Task) Error() error {
+	return task.err
+}
+
+func (task *Task) run(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ttxid, err := task.pastelClient.TicketOwnership(ctx, task.Ticket.Txid, task.Ticket.PastelID, task.Ticket.PastelIDPassphrase)
+	if err != nil {
+		return err
```

Comment:
could you wrap error with context error to easy to tracking? like `return nil, errors.Errorf("....", err)

Reply:
updated. thanks.